python/xoscar/collective/core.py

```diff
@@ -264,60 +391,82 @@ async def gather(
     recv_data: Any,
     root: Optional[int] = 0,
     tag: Optional[int] = 0,
+    stream: Optional[int] = 0,
     group_name: str = "default",
 ):
     address = get_rank_address_via_env(RANK_ADDRESS_ENV_KEY, INVOKE_ERROR_MESSAGE)
     ref = await actor_ref(address=address, uid=f"RankActor")
-    await ref.gather(send_data, recv_data, root=root, tag=tag, pg_name=group_name)
+    await ref.gather(
+        send_data, recv_data, root=root, tag=tag, stream=stream, pg_name=group_name
+    )
 
 
 async def allgather(
-    send_data: Any, recv_data: Any, tag: Optional[int] = 0, group_name: str = "default"
+    send_data: Any,
+    recv_data: Any,
+    tag: Optional[int] = 0,
+    stream: Optional[int] = 0,
+    group_name: str = "default",
 ):
     address = get_rank_address_via_env(RANK_ADDRESS_ENV_KEY, INVOKE_ERROR_MESSAGE)
     ref = await actor_ref(address=address, uid=f"RankActor")
-    await ref.allgather(send_data, recv_data, tag=tag, pg_name=group_name)
+    await ref.allgather(
+        send_data, recv_data, tag=tag, stream=stream, pg_name=group_name
+    )
 
 
 async def scatter(
     send_data: List[Any],
     recv_data: Any,
     root: Optional[int] = 0,
     tag: Optional[int] = 0,
+    stream: Optional[int] = 0,
     group_name: str = "default",
 ):
     address = get_rank_address_via_env(RANK_ADDRESS_ENV_KEY, INVOKE_ERROR_MESSAGE)
     ref = await actor_ref(address=address, uid=f"RankActor")
-    await ref.scatter(send_data, recv_data, root=root, tag=tag, pg_name=group_name)
+    await ref.scatter(
+        send_data, recv_data, root=root, tag=tag, stream=stream, pg_name=group_name
+    )
 
 
 async def reduce_scatter(
     send_data: Any,
     recv_data: Any,
     recv_elems: List[int],
     op: CollectiveReduceOp = CollectiveReduceOp.SUM,
+    stream: Optional[int] = 0,
     group_name: str = "default",
 ):
     address = get_rank_address_via_env(RANK_ADDRESS_ENV_KEY, INVOKE_ERROR_MESSAGE)
     ref = await actor_ref(address=address, uid=f"RankActor")
-    await ref.reduce_scatter(send_data, recv_data, recv_elems, op, pg_name=group_name)
+    await ref.reduce_scatter(
+        send_data, recv_data, recv_elems, op, stream=stream, pg_name=group_name
+    )
 
 
 async def alltoall(
-    send_data: Any, recv_data: Any, tag: Optional[int] = 0, group_name: str = "default"
+    send_data: Any,
+    recv_data: Any,
+    tag: Optional[int] = 0,
+    stream: Optional[int] = 0,
+    group_name: str = "default",
 ):
     address = get_rank_address_via_env(RANK_ADDRESS_ENV_KEY, INVOKE_ERROR_MESSAGE)
     ref = await actor_ref(address=address, uid=f"RankActor")
-    await ref.alltoall(send_data, recv_data, tag=tag, pg_name=group_name)
+    await ref.alltoall(send_data, recv_data, tag=tag, stream=stream, pg_name=group_name)
```

Comment:
Can you provide an example about how to setting this stream in UT?

Reply:
Sure, I provide an example at `test_allreduce` in `test_nccl.py`